test/unittests/parser_test.cpp

```diff
@@ -198,7 +226,8 @@ TEST(parser, type_section_smaller_than_expected)
     const auto section_contents = "01"_bytes + functype_void_to_void + "fe"_bytes;
     const auto bin =
         bytes{wasm_prefix} +
-        make_invalid_size_section(1, size_t{section_contents.size() + 1}, section_contents);
+        make_invalid_size_section(1, size_t{section_contents.size() + 1}, section_contents) +
+        "00"_bytes;
```

Comment:
why this change?

Reply:
We construct a section that claim that has size bigger that its actual content. But to keep the current error we need to add additional byte to the whole wasm binary, because the `expected_section_end` is now checked for out-of-bounds access up front.